distribution/registry.go

```diff
@@ -16,6 +16,8 @@ import (
 	"github.com/docker/docker/registry"
 	"github.com/docker/engine-api/types"
 	"golang.org/x/net/context"
+
+	middleware "github.com/docker/docker/api/server/middleware"
```

Comment:
I would like to avoid having this package import `middleware`, if possible. We've been working in the direction of making `distribution` more self-contained, and may even more it to a separate repo in the future.

Perhaps the `UAStringKey` could be defined in the `dockerversion` package. It could even be made internal to this package, and there could be helper functions defined there to push a user agent onto a context and get the user agent value from a context. This would abstract the context operations so `distribution` and `middleware` don't need to know how a user agent is stored within a context.